### PR TITLE
[Azure] Add sovereign cloud support for MySQL

### LINKED ISSUE #858: [Azure] Add sovereign cloud support for MySQL
The Base URI is incorrectly resolved for the MySQL client, and should be fixed to read from the environment.

Addresses #811

### PR DESCRIPTION
**Pull request by davidkpiano**
This PR resolves the Base URI for the MySQL module to support sovereign cloud.

Addresses #811, fixes #858 

cc. @rguthriemsft

### CODE REVIEW

**Comment by hattan** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/aztfmod/terratest/actions/runs/730444481

**Review by HadwaAbdelhalem — CHANGES_REQUESTED** [review]
Mostly LGTM. One minor NIT change

File: modules/azure/mysql.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 )
 
 // GetMYSQLServerClientE is a helper function that will setup a mysql server client.
+// TODO: remove in next version
```
[the remark this answers is not in the document]
Comment (HadwaAbdelhalem) [reply modules/azure/mysql.go:12]: please ignore this comment, as per another conversation https://github.com/gruntwork-io/terratest/pull/868#discussion_r615078590 , all methods would be removed in one final PR for backward compatibility

**Comment by hattan** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/aztfmod/terratest/actions/runs/768301876

**Review by HadwaAbdelhalem — APPROVED** [review]
LGTM, @gruntwork-io/gruntwork-terratest-terragrunt ready for your review

**Comment by yorinasub17** [comment]
NOTE: Looks like there is a merge conflict now that we merged some of the other PRs.

**Comment by davidkpiano** [comment]
Superseded by #884

**Pull request closed without merging**